docs/examples/using-deno-kv-oauth.md

```diff
@@ -0,0 +1,159 @@
+   instance is stored in `utils/oauth-client.ts`.
+
+```typescript
+// utils/oauth-client.ts
+import { createGitHubOAuth2Client } from "https://deno.land/x/deno_kv_oauth/mod.ts";
+
+export const oauth2Client = createGitHubOAuth2Client();
+```
+
+3. Using the OAuth 2.0 client instance, insert the authentication flow functions
+   into your authentication routes. In this example, there are dedicated handler
+   routes at `routes/auth/signin.ts`, `routes/auth/signout.ts` and
+   `routes/auth/callback.ts`. Please ensure that the `callback` handler matches
+   the Authorization callback URL in the configured OAuth application!
+
+```typescript
+// routes/auth/signin.ts
+import type { Handlers } from "$fresh/server.ts";
+import { signIn } from "https://deno.land/x/deno_kv_oauth/mod.ts";
+import { oauth2Client } from "../../utils/oauth-client.ts";
+
+export const handler: Handlers = {
+  async GET(req) {
+    return await signIn(req, oauth2Client);
+  },
+};
+```
+
+```typescript
+// routes/auth/signout.ts
+import type { Handlers } from "$fresh/server.ts";
+import { signOut } from "https://deno.land/x/deno_kv_oauth/mod.ts";
+
+export const handler: Handlers = {
+  async GET(req) {
+    return await signOut(req, "/");
+  },
+};
+```
+
+```typescript
+// routes/auth/callback.ts
+import type { Handlers } from "$fresh/server.ts";
+import { handleCallback } from "https://deno.land/x/deno_kv_oauth/mod.ts";
+import { oauth2Client } from "../../utils/oauth-client.ts";
+
+export const handler: Handlers = {
+  async GET(req) {
+    // `accessToken` and `sessionId` can be used for further consumption
+    const { accessToken, response, sessionId } = await handleCallback(
+      req,
+      oauth2Client,
+      "/", // an optional `redirectUrl` can be passed
+    );
+
+    return response;
+  },
+};
+```
+
+4. Use Deno KV OAuth's helper functions where needed. These could be part of a
+   `_middleware.ts` handler.
+
+```typescript
+// routes/_middleware.ts
+import type { MiddlewareHandlerContext } from "$fresh/server.ts";
+import {
+  getSessionAccessToken,
+  getSessionId,
+} from "https://deno.land/x/deno_kv_oauth/mod.ts";
+import { oauth2Client } from "../utils/oauth-client.ts";
+
+interface State {
+  session: string | null;
+}
+
+export async function handler(
+  req: Request,
+  ctx: MiddlewareHandlerContext<State>,
+) {
+  const sessionId = await getSessionId(req);
+  // example: use `!!ctx.state.session` to toggle `signin`/`signout` states.
+  ctx.state.session = sessionId;
+
+  if (sessionId != null) {
+    // example: use the `sessionId` to get the `accessToken` to fetch the user from GitHub.
+    const accessToken = await getSessionAccessToken(oauth2Client, sessionId);
+    const response = await fetch("https://api.github.com/user", {
+      headers: {
+        Authorization: `token ${accessToken}`,
+      },
+    });
+    const user = await response.json();
+
+    console.log(user);
+  }
+
+  return await ctx.next();
+}
+```
```

Comment:
Can we instead put this into the home route `/`?

Reply:
absolutely! added a more useful example here.